starboard/nplb/posix_compliance/posix_time_clock_gettime_test.cc

```diff
@@ -0,0 +1,365 @@
+      << "Initial clock_gettime(CLOCK_MONOTONIC) failed. errno: " << errno;
+  int64_t previousMonotonicTime = TimespecToMicroseconds(ts_prev);
+
+  for (int trial = 0; trial < kTrials; ++trial) {
+    struct timespec ts_curr;
+    memset(&ts_curr, 0, sizeof(ts_curr));
+    ASSERT_EQ(0, clock_gettime(CLOCK_MONOTONIC, &ts_curr))
+        << "clock_gettime(CLOCK_MONOTONIC) in loop failed for trial " << trial
+        << ". errno: " << errno;
+    int64_t currentMonotonicTime = TimespecToMicroseconds(ts_curr);
+
+    EXPECT_GE(currentMonotonicTime, previousMonotonicTime)
+        << "clock_gettime(CLOCK_MONOTONIC)-based time should be "
+           "non-decreasing. Previous: "
+        << previousMonotonicTime << ", Current: " << currentMonotonicTime
+        << " on trial " << trial;
+
+    if (currentMonotonicTime < previousMonotonicTime) {
+      FAIL()
+          << "clock_gettime(CLOCK_MONOTONIC)-based time DECREASED. Previous: "
+          << previousMonotonicTime << ", Current: " << currentMonotonicTime
+          << " on trial " << trial;
+      return;
+    }
+    previousMonotonicTime = currentMonotonicTime;
+  }
+  SUCCEED() << "clock_gettime(CLOCK_MONOTONIC)-based time maintained "
+               "non-decreasing behavior over "
+            << kTrials << " trials.";
+}
+
+TEST(PosixTimeClockGettimeTests, ClockRealtimeSucceeds) {
+  struct timespec ts;
+  memset(&ts, 0, sizeof(ts));
+  errno = 0;
+  EXPECT_EQ(0, clock_gettime(CLOCK_REALTIME, &ts))
+      << "clock_gettime(CLOCK_REALTIME) failed. errno: " << errno;
+  EXPECT_GE(ts.tv_nsec, 0);
+  EXPECT_LT(ts.tv_nsec, 1'000'000'000L);
+}
+
+TEST(PosixTimeClockGettimeTests, ClockProcessCpuTimeSucceedsAndIncreases) {
+  struct timespec ts_before, ts_after;
+  memset(&ts_before, 0, sizeof(ts_before));
+  memset(&ts_after, 0, sizeof(ts_after));
+
+  errno = 0;
+  int ret_before = clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts_before);
+  if (ret_before == -1 && errno == EINVAL) {
+    GTEST_SKIP() << "CLOCK_PROCESS_CPUTIME_ID not supported on this system "
+                    "(returned EINVAL).";
+    return;
+  }
+  ASSERT_EQ(0, ret_before)
+      << "Initial clock_gettime(CLOCK_PROCESS_CPUTIME_ID) failed. errno: "
+      << errno;
+
+  // Consume some CPU time.
+  const int kCpuWorkIterations = 2000000;
+  std::atomic<int> counter(0);
+  for (int i = 0; i < kCpuWorkIterations; ++i) {
+    counter++;
+  }
+
+  errno = 0;
+  ASSERT_EQ(0, clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts_after))
+      << "Final clock_gettime(CLOCK_PROCESS_CPUTIME_ID) failed. errno: "
+      << errno;
+
+  int64_t time_before_us = TimespecToMicroseconds(ts_before);
+  int64_t time_after_us = TimespecToMicroseconds(ts_after);
+
+  EXPECT_GE(time_after_us, time_before_us)
+      << "Process CPU time should be non-decreasing. Before: " << time_before_us
+      << " us, After: " << time_after_us << " us.";
+}
+
+TEST(PosixTimeClockGettimeTests, ClockThreadCpuTimeSucceedsAndIncreases) {
```

Comment:
Is this exactly the same as above besides the CLOCK type? Should we be using TEST_P?